src/scheduler/job_info.c

```diff
@@ -1093,6 +1113,13 @@ query_job(struct batch_status *job, server_info *sinfo, schd_error *err)
 				}
 			}
 		}
+		else if (!strcmp(attrp -> name, ATTR_rel_l)) {
+			resreq = find_alloc_resource_req_by_str(resresv -> job -> resreq_rel, attrp -> resource);
+			if (resreq != NULL)
+				set_resource_req(resreq, attrp -> value);
+			if (resresv -> job -> resreq_rel == NULL)
+				resresv -> job -> resreq_rel = resreq;
```

Comment:
I think it is okay, find_resource_req_by_str is going to allocate one if it the resource name is not found in the array. It can return only when there is malloc failure and we will be setting NULL to resreq_rel which was already NULL. So it won't make a difference.